spec/porky_lib/file_service_spec.rb

```diff
@@ -94,240 +95,384 @@ def write_test_file(data)
     tempfile
   end
 
-  it 'write encrypted data to S3' do
-    file_key = file_service.write(plaintext_data, bucket_name, default_key_id)
-    expect(file_key).not_to be_nil
-  end
+  describe '#write' do
+    it 'raises FileServiceError when file is nil' do
+      expect do
+        file_service.write(nil, bucket_name, default_key_id)
+      end.to raise_exception(PorkyLib::FileService::FileServiceError)
+    end
 
-  it 'write large encrypted data to S3' do
-    file_key = file_service.write(File.read("spec#{File::SEPARATOR}porky_lib#{File::SEPARATOR}data#{File::SEPARATOR}large_plaintext"),
-                                  bucket_name, default_key_id)
-    expect(file_key).not_to be_nil
-  end
+    it 'raises FileServiceError when bucket name is nil' do
+      expect do
+        file_service.write(plaintext_data, nil, default_key_id)
+      end.to raise_exception(PorkyLib::FileService::FileServiceError)
+    end
 
-  it 'write encrypted data to S3 with metadata' do
-    metadata = { content_type: 'test/data' }
-    file_key = file_service.write(plaintext_data, bucket_name, default_key_id, metadata: metadata)
-    expect(file_key).not_to be_nil
-  end
+    it 'raises FileServiceError when key ID is nil ' do
+      expect do
+        file_service.write(plaintext_data, bucket_name, nil)
+      end.to raise_exception(PorkyLib::FileService::FileServiceError)
+    end
 
-  it 'write encrypted data to S3 with directory' do
-    file_key = file_service.write(plaintext_data, bucket_name, default_key_id, directory: '/directory1/dirA')
-    expect(file_key).not_to be_nil
-  end
+    # rubocop:disable RSpec/MessageSpies
```

Comment:
Needed to disable rubocop, since the way it suggested (with `have_received`) made it so that the tempfile was already deleted from the system and cannot be read

Reply:
come on rubocop! 😅